File: ios/Classes/AviaryTunnelPlugin.swift
import Flutter
import UIKit

public class AviaryTunnelPlugin: NSObject, FlutterPlugin, MessengerHostApi {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let instance = AviaryTunnelPlugin()
        MessengerHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: instance)
    }
    
    func sendMessage(message: FlutterStandardTypedData?, completion: @escaping (Result<FlutterStandardTypedData?, Error>) -> Void) {
        let message = Message.with { (message) in
            message.subject = "platform"
            message.body = "iOS " + UIDevice.current.systemVersion + " by Pigeon with Protobuf"
        }
        guard let data = try? message.serializedData() else {
            return
        }
        completion(.success(FlutterStandardTypedData(bytes: data)))
    }
}

File: android/src/main/kotlin/org/aghnyap/aviary/aviary_tunnel/AviaryTunnelPlugin.kt
package org.aghnyap.aviary.aviary_tunnel

import MessageOuterClass.Message
import MessengerHostApi

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** AviaryTunnelPlugin */
class AviaryTunnelPlugin: FlutterPlugin, MessengerHostApi {
  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    MessengerHostApi.setUp(flutterPluginBinding.binaryMessenger, this)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    MessengerHostApi.setUp(binding.binaryMessenger, null)
  }

  override fun sendMessage(message: ByteArray?, callback: (Result<ByteArray?>) -> Unit) {
    val message = Message.newBuilder().apply {
      subject = "platform"
      body = "Android ${android.os.Build.VERSION.RELEASE} by Pigeon with Protobuf"
    }.build()

    callback(Result.success(message.toByteArray()))
  }
}
